driver/src/main/java/oracle/nosql/driver/ops/QueryRequest.java

```diff
@@ -451,14 +456,50 @@ public QueryRequest setMaxMemoryConsumption(long maxBytes) {
      * array or map) and sorting (sorting by distance when a query contains
      * a geo_near() function). Such operations may consume a lot of memory
      * as they need to cache the full result set at the client memory.
-     * The default value is 100MB.
+     * <p>
+     * The default value is 1GB.
      *
      * @return the maximum number of memory bytes
      */
     public long getMaxMemoryConsumption() {
         return maxMemoryConsumption;
     }
 
+    /**
+     * Sets the maximum number of memory bytes that may be consumed by the
+     * statement at a replication node. In general, queries do not consume
+     * a lot of memory while executing at a replcation node and the value
+     * of this parameter has no effect. Currently, the only exception are
+     * queries that use the array_collect function. For such queries, if
+     * the maximum amount of memory is exceeded, execution of the query
+     * at the replication node will be terminated (without error) and the
+     * set of query results that have been computed so far will be sent
+     * to the driver. The driver will keep executing the query, sending 
+     * more requests to the replication nodes for additional results.
+     * So, for queries that use array_collect, increasing the value of this
+     * parameter will decrease the number of interactions between the driver
+     * and the replacation nodes at the expense of consuming the memory
+     * consumption at the nodes.
+     * <p> 
+     * The default value is 10MB, and for applacations running on the
+     * cloud, it can not be inreased beyond this default.
```

Comment:
The default value of 10MB is the max value of maxServerMemoryConsumption, it is best configuration to reduce the requests between driver and RN. Under what circumstances is it recommended that users use smaller maxServerMemoryConsumption？

Typo: applacations -> applications, inreased -> increased

Reply:
I have fixed the typos.
After a discussion with Dave, I have marked both this and the getter method as hidden. This is because this param is too esoteric and we don't really expect users to change it. Furthermore, it does not really apply to the cloud, because in the cloud it cannot increased, and as you say, there is no reason to decrease it.